hoodie-client/src/main/java/com/uber/hoodie/index/HoodieBloomIndex.java

```diff
@@ -151,7 +151,13 @@ public HoodieBloomIndex(HoodieWriteConfig config, JavaSparkContext jsc) {
         for (String partitionPath : recordsPerPartition.keySet()) {
             long numRecords = recordsPerPartition.get(partitionPath);
             long numFiles = filesPerPartition.containsKey(partitionPath) ? filesPerPartition.get(partitionPath) : 1L;
-            subpartitionCountMap.put(partitionPath, ((numFiles * numRecords) / MAX_ITEMS_PER_JOIN_PARTITION) + 1);
+            // minimum number of sub partitions we need to avoid a single sub partition reading more than config.getMaxIndexFileLookup()
+            long minPartitionsForMaxFileRead = (numFiles / config.getMaxIndexFileLookup()) + 1;
```

Comment:
So the theory is to overpartition and see if its faster? I thought about this back then. There is the additional cost of re-reading bloom filters yet again due to records being split across partitions.. 

.. I guess we could bet on HDFS parallel reads of same data to be faster.. Anyhow, good to have this configurable.. Lets see how things work out.